polaris-core/src/main/java/io/polaris/core/storage/PolarisStorageConfigurationInfo.java

```diff
@@ -187,7 +188,7 @@ public static Optional<PolarisStorageConfigurationInfo> forEntityPath(
 
   /** Validate if the provided allowed locations are valid for the storage type */
   protected void validatePrefixForStorageType(String loc) {
-    if (!loc.toLowerCase().startsWith(storageType.prefix)) {
+    if (!loc.toLowerCase(Locale.ROOT).startsWith(storageType.prefix)) {
```

Comment:
How about introducing a [modernizer](https://github.com/andygoossens/gradle-modernizer-plugin), errorprone's [StringCaseLocaleUsage](https://errorprone.info/bugpattern/StringCaseLocaleUsage) rule or something to find this usage automatically?